### PR TITLE
Add offline_mirror_url parameter

### PR DESCRIPTION
**Pull request by Bialogs**
#### Pull Request (PR) description
Add `$offline_mirror_url` parameter to support arbitrary offline filenames. 

When defined, this url will override any other download location.

#### This Pull Request (PR) fixes the following issues
This may address a portion of #38 (at least it addresses my concerns).

### CODE REVIEW

**Comment by bastelfreak** [comment]
Hi @Bialogs, thanks for the PR! Can you please take a look at the failing spec tests and add unit tests for the new param?

**Comment by bastelfreak** [comment]
hrm we need https://github.com/voxpupuli/puppet-kafka/pull/275 to get travis green again

**Comment by Bialogs** [comment]
Are the failures only related to #275?

**Comment by bastelfreak** [comment]
@Bialogs the failures we see on travis are related to #275. They may hide other errors.

**Comment by Bialogs** [comment]
Thanks I'll be paying attention to see when that gets merged. 

In the meantime, if you have any feedback on the unit test I added I'd love to hear it; I'm new to `rspec` and `rspec-puppet`.

File: spec/classes/init_spec.rb [hunk 1]
```diff
@@ -59,4 +59,20 @@
...
+    describe 'kafka' do
+      context 'offline mirror' do
+        let :params do
+          {
+            offline_mirror_url: 'https://foobar.com/another/kafka'
+          }
+        end
+
+        it {
+          is_expected.to contain_archive('/var/tmp/kafka/kafka_2.11-0.11.0.3.tgz').with(source: 'https://foobar.com/another/kafka')
```
Comment (dhoppe) [inline spec/classes/init_spec.rb]: The version of Kafka and Scala has changed:
* https://github.com/voxpupuli/puppet-kafka/pull/291

**Pull request closed without merging**